packages/adp-tooling/src/base/change-utils.ts

```diff
@@ -165,11 +165,11 @@ export function findChangeWithInboundId(projectPath: string, inboundId: string):
     }
 
     try {
-        const file: Dirent | undefined = readdirSync(pathToInboundChangeFiles, { withFileTypes: true }).find(
-            (dirent: Dirent) => dirent.isFile() && dirent.name.includes('changeInbound')
+        const files: Dirent[] = readdirSync(pathToInboundChangeFiles, { withFileTypes: true }).filter(
+            (dirent) => dirent.isFile() && dirent.name.includes('changeInbound')
         );
```

Comment:
Can we use this approach for getting the needed file e.g something like this:
`const file = readdirSync(pathToInboundChangeFiles, { withFileTypes: true }).filter(
			(dirent) => dirent.isFile() && dirent.name.includes('changeInbound')).map(function(file) {
				const filePath = join(pathToInboundChangeFiles, file.name);
				return {
					changeObject: JSON.parse(readFileSync(filePath, 'utf-8')),
					filePath: filePath
				}
			}).find((element) => element.changeObject?.change?.inboundId === inboundId);`

Reply:
Can you format this please?

Reply:
I am against this approach because it will make it less readable, maintainable, and potentially less efficient.

Reply:
Added a break statement here performance: [dbb6980](https://github.com/SAP/open-ux-tools/pull/1745/commits/dbb698046df6ab7a9981aa070b00889e08ac8635)